File: backend/src/main/java/com/example/demo/service/FundingService.java
package com.example.demo.service;

import com.example.demo.dto.request.FundingRequestDTO;
import com.example.demo.dto.response.FundingResponseDTO;

import java.util.List;

public interface FundingService {

    FundingResponseDTO createFunding(FundingRequestDTO request);

    FundingResponseDTO getFundingById(Long id);

    List<FundingResponseDTO> getAllFunding();

    FundingResponseDTO updateFunding(Long id, FundingRequestDTO request);

    void deleteFunding(Long id);
}

File: frontend/src/services/fundingService.js
import api from "../api/api";

export const getAllFunding = async () => {
    const response = await api.get("/funding");
    return response.data;
};

export const getFundingById = async (id) => {
    const response = await api.get(`/funding/${id}`);
    return response.data;
};

export const createFunding = async (data) => {
    const response = await api.post("/funding", data);
    return response.data;
};

export const updateFunding = async (id, data) => {
    const response = await api.put(`/funding/${id}`, data);
    return response.data;
};

export const deleteFunding = async (id) => {
    const response = await api.delete(`/funding/${id}`);
    return response.data;
};
